src/utils/response/responseHandlers.ts

```diff
@@ -1,58 +1,53 @@
-    const dataPromise = await promise;
-    const apiResponse = (dataPromise.data as unknown) as ApiResponse<T>;
+type RequestPromise<T> = Promise<AxiosResponse<ApiResponse<T>>>;
 
-    if (apiResponse.Success) {
-      return apiResponse;
-    }
+export async function toApiResponse<T>(promise: RequestPromise<T>) {
+  const apiResponse = (await promise).data;
 
-    throw apiResponse;
+  if (apiResponse.Success) {
+    return apiResponse;
   }
 
-  /**
-   * @return A promise that resolves to T or rejects if no results
-   */
-  required<T>(promise: Promise<AxiosResponse>): Promise<T> {
-    return this.toApiResponse<T[]>(promise).then((response: ApiResponseOnSuccess<T[]>) => {
-      if (response.NumResults > 0) {
-        return response.Results[0];
-      } else {
-        throw new Error('There are no results.');
-      }
-    });
-  }
+  throw apiResponse;
+}
 
-  requiredList<T>(promise: Promise<AxiosResponse>): Promise<T[]> {
-    return this.toApiResponse<T[]>(promise).then((response: ApiResponseOnSuccess<T[]>) => {
-      if (response.NumResults > 0) {
-        return response.Results;
-      } else {
-        throw new Error('There are no results.');
-      }
-    });
-  }
+/**
+ * @return A promise that resolves to T or rejects if no results
+ */
+export async function required<T>(promise: RequestPromise<T[]>): Promise<T> {
+  const response = await toApiResponse<T[]>(promise);
 
-  /**
-   * @return A promise that resolves to T or undefined if no results but Success is true.
-   */
-  optional<T>(promise: Promise<AxiosResponse>): Promise<T | undefined> {
-    return this.toApiResponse<T[]>(promise).then((response: ApiResponseOnSuccess<T[]>) => {
-      return (response.NumResults > 0 && response.Results[0]) || undefined;
-    });
+  if (response.NumResults > 0) {
+    return response.Results[0];
+  } else {
+    throw new Error('There are no results.');
   }
+}
 
-  /**
-   * @return A promise that resolves to array of T or null if no results but Success is true.
-   */
-  optionalList<T>(promise: Promise<AxiosResponse>): Promise<T[]> {
-    return this.toApiResponse<T[]>(promise).then((response: ApiResponseOnSuccess<T[]>) => {
-      return (response.NumResults > 0 && response.Results) || [];
-    });
+export async function requiredList<T>(promise: RequestPromise<T[]>): Promise<T[]> {
+  const response = await toApiResponse<T[]>(promise);
+
+  if (response.NumResults > 0) {
+    return response.Results;
+  } else {
+    throw new Error('There are no results.');
   }
 }
 
-export default new ResponseHandler();
+/**
+ * @return A promise that resolves to T or undefined if no results but Success is true.
+ */
+export async function optional<T>(promise: RequestPromise<T[]>): Promise<T | undefined> {
+  const response = await toApiResponse<T[]>(promise);
+  return (response.NumResults > 0 && response.Results[0]) || undefined;
+}
+
+/**
+ * @return A promise that resolves to array of T or null if no results but Success is true.
```

Comment:
Out of scope in this PR, but having this resolve to "array of T or null", while `optional()` resolves to "T or undefined" seems like something that might come back and bite us in one place in the future.

Also, unless I'm missing something, it's not even an "array of null" -- just an empty array.

Reply:
yeah, that comment is wrong, this just returns an array, never null or undefined. Not sure why we even check for NumResults, we should be able to just return response.Results as-is. Also I'm not sure whether requiredList is even needed and I have the feeling that it might be in use in some places where optionalList should be used. Ideally we can remove requiredList and have optionalList be just "list".  Probably would be best to create a ticket for that